### PR TITLE
Adaptive step-size for ```FineRKN5```

### PR DESCRIPTION
**Pull request by HenryLangner**
The changes enable the use of adaptive step sizes for ```FineRKN5()```. 
The already implemented convergence tests are all passed. I will add convergence tests for the damped oscillator introduced in https://github.com/SciML/OrdinaryDiffEq.jl/pull/1973, once this PR and https://github.com/SciML/OrdinaryDiffEq.jl/pull/1975 are merged. 

CC @ranocha

### CODE REVIEW

File: src/perform_step/rkn_perform_step.jl [hunk 1]
```diff
@@ -225,8 +237,24 @@ end
...
+        duhat, uhat = utilde.x
+        dtsq = dt^2
+        @tight_loop_macros for i in uidx
+            @inbounds uhat[i] = dtsq *
+                                (btilde1 * k1[i] + btilde3 * k3[i] + btilde4 * k4[i] +
+                                 btilde5 * k5[i])
+            @inbounds duhat[i] = dt *
+                                 (bptilde1 * k1[i] + bptilde3 * k3[i] + bptilde4 * k4[i] +
+                                  bptilde5 * k5[i] + bptilde6 * k6[i] + bptilde7 * k7[i])
+        end
```
Comment (ranocha) [inline src/perform_step/rkn_perform_step.jl:252]: Could you please use braodcasting here?

**Review by ranocha — CHANGES_REQUESTED** [review]
Could you please add some tests with adaptive time stepping, too?

**Comment by HenryLangner** [comment]
> Could you please add some tests with adaptive time stepping, too?

Regression tests for adaptive time stepping are added.

**Review by ranocha — APPROVED** [review]
Thanks!

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**